python-avd/pyavd/_eos_designs/structured_config/network_services/port_channel_interfaces.py

```diff
@@ -25,16 +27,89 @@ def port_channel_interfaces(self: AvdStructuredConfigNetworkServicesProtocol) ->
         """
         Return structured config for port_channel_interfaces.
 
-        Only used with L1 network services
+        Only used with L1 network services or L3 network services
         """
-        if not self.shared_utils.network_services_l1:
+        if not self.shared_utils.network_services_l1 and not self.shared_utils.network_services_l3:
             return None
 
         # Using temp variables to keep the order of interfaces from Jinja
         port_channel_interfaces = []
         subif_parent_interfaces = []
 
         for tenant in self.shared_utils.filtered_tenants:
+            for vrf in tenant.vrfs:
+                # The l3_port_channel has already been filtered in filtered_tenants
+                # to only contain entries with our hostname
+                subif_parent_port_channel_names = {}
+                regular_l3_port_channel_names = {}
+                node_names = set()
+                for l3_port_channel in vrf.l3_port_channels:
+                    for _node_index, node_name in enumerate(l3_port_channel.nodes):
+                        if node_name != self.shared_utils.hostname:
+                            continue
```

Comment:
I am not sure you need this loop if you never use _node_index. We normally use the node index to fish out the ip address etc in the other lists.

Reply:
removed